src/poetry/core/masonry/builders/sdist.py

```diff
@@ -320,17 +321,21 @@ def find_files_to_add(self, exclude_build: bool = False) -> set[BuildIncludeFile
         to_add = super().find_files_to_add(exclude_build)
 
         # add any additional files, starting with all LICENSE files
-        additional_files = set(self._path.glob("LICENSE*"))
+        additional_files: set[str | Path] = set(self._path.glob("LICENSE*"))
 
         # add script files
         additional_files.update(self.convert_script_files())
 
         # Include project files
         additional_files.add(Path("pyproject.toml"))
 
-        # add readme if it is specified
+        # add readme files if it is specified
         if "readme" in self._poetry.local_config:
-            additional_files.add(self._poetry.local_config["readme"])
+            r: str | Path | Iterable[Path] = self._poetry.local_config["readme"]
```

Comment:
This seems even more convoluted -- can you tell me where we are getting a conversion to `Path` from?

Reply:
`BuildIncludeFile` is handling the `str` to `Path` conversion

Reply:
In addition, I just confirmed `local_config["readme"]` is actually `str` only (or `Iterable[str]`.

I added `print(type(local_config["readme"][0]))` statement to `Poetry` constructor on my patched poetry-core lib and run `poetry build` on a project sample with 2 readme files.

The result was `<class 'tomlkit.items.String'>`, which is a subclass of `str`, declared as `class String(str, Item)`.

Reply:
I pushed a change typing `local_config["reade"]` simply as `str | Iterable[str]`